src/model/cloud-runner/aws/aws-task-runner.ts

```diff
@@ -0,0 +1,216 @@
+import * as AWS from 'aws-sdk';
+import CloudRunnerEnvironmentVariable from '../services/cloud-runner-environment-variable';
+import * as core from '@actions/core';
+import CloudRunnerTaskDef from '../services/cloud-runner-task-def';
+import * as zlib from 'zlib';
+import CloudRunnerLogger from '../services/cloud-runner-logger';
+import { Input } from '../..';
+import fs from 'fs';
+import { CloudRunnerState } from '../state/cloud-runner-state';
+import { CloudRunnerStatics } from '../cloud-runner-statics';
+
+class AWSTaskRunner {
+  static async runTask(
+    taskDef: CloudRunnerTaskDef,
+    ECS: AWS.ECS,
+    CF: AWS.CloudFormation,
+    environment: CloudRunnerEnvironmentVariable[],
+    buildGuid: string,
+    commands: string[],
+  ) {
+    const cluster = taskDef.baseResources?.find((x) => x.LogicalResourceId === 'ECSCluster')?.PhysicalResourceId || '';
+    const taskDefinition =
+      taskDef.taskDefResources?.find((x) => x.LogicalResourceId === 'TaskDefinition')?.PhysicalResourceId || '';
+    const SubnetOne =
+      taskDef.baseResources?.find((x) => x.LogicalResourceId === 'PublicSubnetOne')?.PhysicalResourceId || '';
+    const SubnetTwo =
+      taskDef.baseResources?.find((x) => x.LogicalResourceId === 'PublicSubnetTwo')?.PhysicalResourceId || '';
+    const ContainerSecurityGroup =
+      taskDef.baseResources?.find((x) => x.LogicalResourceId === 'ContainerSecurityGroup')?.PhysicalResourceId || '';
+    const streamName =
+      taskDef.taskDefResources?.find((x) => x.LogicalResourceId === 'KinesisStream')?.PhysicalResourceId || '';
+
+    const task = await ECS.runTask({
+      cluster,
+      taskDefinition,
+      platformVersion: '1.4.0',
+      overrides: {
+        containerOverrides: [
+          {
+            name: taskDef.taskDefStackName,
+            environment: [...environment, { name: 'BUILDID', value: buildGuid }],
+            command: ['-c', ...commands],
+          },
+        ],
+      },
+      launchType: 'FARGATE',
+      networkConfiguration: {
+        awsvpcConfiguration: {
+          subnets: [SubnetOne, SubnetTwo],
+          assignPublicIp: 'ENABLED',
+          securityGroups: [ContainerSecurityGroup],
+        },
+      },
+    }).promise();
+
+    CloudRunnerLogger.log('Cloud runner job is starting');
+    const taskArn = task.tasks?.[0].taskArn || '';
```

Comment:
I have no idea what `taskArn` means. Prefer more descriptive variable name. Unless this is a really common term.

Reply:
Arn is like a funny AWS id name.

I think arn is like a resource notation.

Reply:
https://docs.aws.amazon.com/general/latest/gr/aws-arns-and-namespaces.html
yep

Reply:
they should use more plain English and fewer acronyms :P